frontend/src/components/MapView/DateSelector/index.tsx

```diff
@@ -93,6 +93,7 @@ function DateSelector({
   const [pointerPosition, setPointerPosition] = useState<Point>({ x: 0, y: 0 });
 
   const dateRef = useRef(availableDates);
+  const maxDate = new Date(Math.max(...availableDates, new Date().getTime()));
```

Comment:
I think it would be good to memoize it using the following so that we will have a cached value through rerenders if the available dates array is not changed.
```suggestion
    const maxDate = useMemo(() => {
    return new Date(Math.max(...availableDates, new Date().getTime()));
  }, [availableDates]);
```